webhooks/collector.go

```diff
@@ -23,148 +22,75 @@ func (webhookList *WebhookList) AddItem(item WebhookMessage) {
 	webhookList.Messages = append(webhookList.Messages, item)
 }
 
-var webhookCollections map[string]*WebhookList
-
-const GymDetails = "gym_details"
-const Raid = "raid"
-const Pokemon = "pokemon"
-const Quest = "quest"
-const Pokestop = "pokestop"
-const Invasion = "invasion"
-const Weather = "weather"
-const FortUpdate = "fort_update"
+const (
+	GymDetails  = config.WebhookTypeGymDetails
+	Raid        = config.WebhookTypeRaid
+	Quest       = config.WebhookTypeQuest
+	Pokestop    = config.WebhookTypePokestop
+	Invasion    = config.WebhookTypeInvasion
+	Weather     = config.WebhookTypeWeather
+	FortUpdate  = config.WebhookTypeFortUpdate
+	PokemonIV   = config.WebhookTypePokemonIV
+	PokemonNoIV = config.WebhookTypePokemonNoIV
+)
 
+var webhookCollections map[config.WebhookType]*WebhookList
 var collectionAccess sync.Mutex
 
 func init() {
-	SetMaps()
-
+	initCollections()
 }
 
-func SetMaps() {
-	webhookCollections = make(map[string]*WebhookList)
+func initCollections() {
+	webhookCollections = make(map[config.WebhookType]*WebhookList)
 	webhookCollections[GymDetails] = &WebhookList{}
 	webhookCollections[Raid] = &WebhookList{}
-	webhookCollections[Pokemon] = &WebhookList{}
+	webhookCollections[PokemonIV] = &WebhookList{}
+	webhookCollections[PokemonNoIV] = &WebhookList{}
 	webhookCollections[Quest] = &WebhookList{}
 	webhookCollections[Pokestop] = &WebhookList{}
 	webhookCollections[Invasion] = &WebhookList{}
 	webhookCollections[Weather] = &WebhookList{}
 	webhookCollections[FortUpdate] = &WebhookList{}
 }
 
-func AddMessage(webhookType string, message interface{}, areas []geo.AreaName) {
-	collectionAccess.Lock()
-	list := webhookCollections[webhookType]
-	list.AddItem(WebhookMessage{
-		Type:    webhookType,
+func AddMessage(webhookType config.WebhookType, message any, areas []geo.AreaName) {
+	wh_message := WebhookMessage{
+		Type:    webhookType.PayloadType(),
 		Areas:   areas,
 		Message: message,
-	})
-	//webhookCollections[webhookType] = list
+	}
+	collectionAccess.Lock()
+	webhookCollections[webhookType].AddItem(wh_message)
 	collectionAccess.Unlock()
 }
 
 func collectHooks() []WebhookQueue {
 	collectionAccess.Lock()
 	currentCollection := webhookCollections
-	SetMaps()
+	initCollections()
 	collectionAccess.Unlock()
 
 	var destinations []WebhookQueue
 
 	for _, hook := range config.Config.Webhooks {
```

Comment:
It is difficult to tell by reading; can you confirm that this still works where is no list specified at all (ie null before), in which case all webhooks are sent?

Reply:
Good question. Yes. In webhook.initAfterLoad(), if there are no types defined in the config, it copies all possibilities into .typesWanted such that they'll be returned by the call to TypesWanted().